src/rollup/types.d.ts

```diff
@@ -67,12 +72,26 @@ export interface ModuleJSON {
 	resolvedIds: IdMap;
 }
 
+export interface Asset {
+	name: string;
+	source: string | Buffer;
+	fileName: string;
+	transform: boolean;
+	dependencies: string[];
+}
+
 export interface PluginContext {
 	watcher: Watcher;
 	resolveId: ResolveIdHook;
 	isExternal: IsExternal;
 	parse: (input: string, options: any) => ESTree.Program;
-	emitAsset: (name: string, source?: string | Buffer) => string;
+	emitAsset(name: string, source?: string | Buffer): string;
+	emitAsset(name: string, dependencies: string[], source?: string | Buffer): string;
+	emitAsset(
+		name: string,
+		dependenciesOrSource: string[] | string,
+		source?: string | Buffer
+	): string;
```

Comment:
I actually would not put the "effective signature" into the official types as well as this does not reflect how people should use this hook (they should use either of the previous signatures). As everything else looks splendid, I hope you don't mind if I just remove this before merging.